Constant generation for my project broken by PR #3288

Constant generation for my project got broken by PR #3288. I can't copy-paste a repro directly because of closed source blah blah blah, will try to get a minimal repro and post it here in the next week or so. But the general pattern looks like this:

* We have some proprietary C headers and we're running bindgen to generate rust bindings to them
* Headers have constants like `#define FooBar 0x00000000`
* Expectation: bindgen should create a constant like `pub const FooBar: u32 = 0;`. Bindgen actually does that prior to #3288
* Post #3288, bindgen creates this: `pub const FooBar: u32 = u32(0);`
* This fails to compile with the error: `error[E0423]: expected function, found builtin type `u32``
* We're using bindgen as a library, calling it from `build.rs`

Here is the essence of our bindgen config (I tried to remove non-relevant stuff):

```
let mut builder = bindgen::Builder::default()
        .derive_default(true)
        .prepend_enum_name(false)
        .parse_callbacks(Box::new(bindgen::CargoCallbacks::new()))
        .default_enum_style(bindgen::EnumVariation::NewType {
            is_bitfield: false,
            is_global: false,
        })
        .default_alias_style(bindgen::AliasVariation::NewType)
        .generate_cstr(true)
        .no_copy("xxx") // in reality a bunch of regexes
        .type_alias("yyy") // in reality a bunch of regexes
        .merge_extern_blocks(true)
        .derive_debug(false)
        .impl_debug(false)
        .parse_callbacks(Box::new(AddDerivesCallbacks::new())); // these callbacks just add some traits to certain types based on regexes
```

--- a/bindgen-tests/tests/expectations/tests/issue-3303-new-type-alias-default.rs
+++ b/bindgen-tests/tests/expectations/tests/issue-3303-new-type-alias-default.rs
@@ -1,0 +1,2 @@
+#![allow(dead_code, non_snake_case, non_camel_case_types, non_upper_case_globals)]
+pub const BOGUS_CONST: u32 = u32(0);

--- a/bindgen-tests/tests/expectations/tests/issue-3303-new-type-alias-default.rs
+++ b/bindgen-tests/tests/expectations/tests/issue-3303-new-type-alias-default.rs
@@ -1,2 +1,2 @@
 #![allow(dead_code, non_snake_case, non_camel_case_types, non_upper_case_globals)]
-pub const BOGUS_CONST: u32 = u32(0);
+pub const BOGUS_CONST: u32 = 0;

--- a/bindgen-tests/tests/headers/issue-3303-new-type-alias-default.h
+++ b/bindgen-tests/tests/headers/issue-3303-new-type-alias-default.h
@@ -1,0 +1,3 @@
+// bindgen-flags: --default-alias-style=new_type
+
+#define BOGUS_CONST 0x00000000

--- a/bindgen/codegen/mod.rs
+++ b/bindgen/codegen/mod.rs
@@ -810,11 +810,19 @@ impl CodeGenerator for Var {
             };
 
             if let Some(mut val) = const_expr {
-                let var_ty_item = ctx.resolve_item(var_ty);
-                if matches!(
-                    var_ty_item.alias_style(ctx),
-                    AliasVariation::NewType | AliasVariation::NewTypeDeref
-                ) {
+                let var_ty_item =
+                    var_ty.into_resolver().through_type_refs().resolve(ctx);
+
+                let is_alias = var_ty_item
+                    .as_type()
+                    .is_some_and(|ty| matches!(ty.kind(), TypeKind::Alias(..)));
+
+                if is_alias &&
+                    matches!(
+                        var_ty_item.alias_style(ctx),
+                        AliasVariation::NewType | AliasVariation::NewTypeDeref
+                    )
+                {
                     val = quote! { #ty(#val) };
                 }
                 result.push(quote! {

